### PR TITLE
Feat: new ETH stake flow. Modal Kiln and Lido.

### PR DESCRIPTION
**Pull request by sarneijim**
### 📝 Description

Implement new ETH stake flow, new modal
Kiln and Lido.

- Stake bottom from account flow
- Stake banner from receive flow (with not show more checkbox)

Details:
- Flag provider configurable
- Translation are needed
- Account selected in provider iframe

Support new and old flag definition:

- feature_eth_staking_providers:
``
{"enabled":true,"params":{"listProvider":[{"name":"Lido","minAccountBalance":0,"liveAppId":"lido", "supportLink":"https://support.ledger.com/hc/en-us/articles/4406882832401-Staking-ETH-in-Ledger-Live-with-Lido?docs=true"},{"name":"Kiln","minAccountBalance":32,"liveAppId":"kiln", "supportLink": "https://www.kiln.fi/" }]}}
``
- feature_receive_staking_flow_config_desktop
``
{"enabled":true,"params":{"ethereum":{"enabled":true,"supportLink":"https://www.ledger.com/staking-ethereum", "direct": true},"solana":{"enabled":true,"supportLink":"https://www.ledger.com/staking/ledger-node/solana", "direct": false},"cosmos":{"enabled":true,"supportLink":"https://www.ledger.com/staking/ledger-node/cosmos", "direct": false},"celo":{"enabled":true,"supportLink":"https://support.ledger.com/hc/en-us/articles/5046374786333-Earn-rewards-with-Celo-CELO-in-Ledger-Live?docs=true", "direct": false},"osmo":{"enabled":true,"supportLink":"https://www.ledger.com/staking/staking-osmosis", "direct": false}}}
``

Only New flag definition:

- feature_eth_staking_providers:
``
{"enabled":true,"params":{"listProvider":[{"name":"Lido","liveAppId":"lido", "supportLink":"https://support.ledger.com/hc/en-us/articles/4406882832401-Staking-ETH-in-Ledger-Live-with-Lido?docs=true"},{"name":"Kiln","liveAppId":"kiln", "supportLink": "https://www.kiln.fi/" }]}}
``
- feature_receive_staking_flow_config_desktop
``
{"enabled":true,"params":{"ethereum":{"enabled":true,"supportLink":"https://www.ledger.com/staking-ethereum", "direct": true},"solana":{"enabled":true,"supportLink":"https://www.ledger.com/staking/ledger-node/solana", "direct": false},"cosmos":{"enabled":true,"supportLink":"https://www.ledger.com/staking/ledger-node/cosmos", "direct": false},"celo":{"enabled":true,"supportLink":"https://support.ledger.com/hc/en-us/articles/5046374786333-Earn-rewards-with-Celo-CELO-in-Ledger-Live?docs=true", "direct": false},"osmo":{"enabled":true,"supportLink":"https://www.ledger.com/staking/staking-osmosis", "direct": false}}}
``

### ❓ Context

- **Impacted projects**: 
  - LLD
- **Linked resource(s)**: 
  -  LLD - ETH staking button redirecting to banner for Lido and Kiln https://ledgerhq.atlassian.net/browse/LIVE-4668
  - LLD - Receive banner for Lido and Kiln https://ledgerhq.atlassian.net/browse/LIVE-4348

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

Basic modal
Screenshot 2022-12-03 at 18 22 20

Receive modal
Screenshot 2022-12-03 at 18 32 29

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/Body.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    font-size: 14px;
+    line-height: 18px;
+  }
+`;
+
+const Body = ({ checkbox = false, source = "stake", ...itemProps }: Props) => {
+  const { t } = useTranslation();
+  const [doNotShowAgain, setDoNotShowAgain] = useState<boolean>(false);
+  const ethStakingProviders = useFeature("ethStakingProviders");
+  const providers = ethStakingProviders?.params?.listProvider || [];
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/Body.jsx:40]: Would the modal show if (for whatever reason) there are no providers?

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 2]
```diff
@@ -3408,6 +3400,21 @@
       }
     }
   },
+  "ethereum": {
+    "stake": {
+      "title": "Earn Ethereum",
+      "lido": {
+        "tag": "No minimun",
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/static/i18n/en/app.json:3407]: Should be 'minimum' @sarneijim

**Comment by ggilchrist-ledger** [comment]
The Lido and Kiln banners are slightly fuzzy and not as sharp as the rest of the app. Not sure why though 😕 
image

**Comment by ggilchrist-ledger** [comment]
The 'fuzziness' seems to be better depending on the screen and window size, so looks like nothing to do with this feature.

File: apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/index.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import React, { PureComponent } from "react";
+import { Trans } from "react-i18next";
+import Modal, { ModalBody } from "~/renderer/components/Modal";
+import Body from "./Body";
+import TrackPage from "~/renderer/analytics/TrackPage";
+import { Flex } from "@ledgerhq/react-ui";
+import type { Account } from "@ledgerhq/types-live";
+
+type Props = { name: string, account: Account, checkbox?: boolean, source?: string };
+export default class DelegationModal extends PureComponent<Props> {
```
Comment (cjordan-ledger) [inline apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/index.jsx:12]: Could this be a functional component instead of class-based?

**Review by cjordan-ledger — COMMENTED** [review]
Code looks good to me, left one minor non-blocking comment

File: apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/index.jsx (hunk 3 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/families/ethereum/StakeFlowModal/index.jsx:12]: Agree. Done

**Review by cjordan-ledger — APPROVED** [review]
Looks good 🎉

**Comment by ggilchrist-ledger** [comment]
LGTM 
[wonderful](https://media.giphy.com/media/l0HUk3XWstjKCnvuo/giphy.gif)

**Pull request merged**